contracts/integrations/test/framework/assertions/moveStake.ts

```diff
@@ -196,4 +198,54 @@ export function validMoveStakeAssertion(
         },
     });
 }
+
+/**
+ * Returns a FunctionAssertion for `moveStake` which asserts that the transaction reverts with a
+ * PoolExistenceError.
+ */
+export function moveStakeNonexistentPoolAssertion(
+    deployment: DeploymentManager,
+    nonExistentPoolId: string,
+): FunctionAssertion<[StakeInfo, StakeInfo, BigNumber], void, void> {
+    return new FunctionAssertion<[StakeInfo, StakeInfo, BigNumber], void, void>(
+        deployment.staking.stakingWrapper,
+        'moveStake',
+        {
+            after: async (_beforeInfo: void, result: FunctionResult) => {
+                // Ensure that the tx reverted.
+                expect(result.success).to.be.false();
+
+                // Check revert error
+                expect(result.data).to.equal(new StakingRevertErrors.PoolExistenceError(nonExistentPoolId, false));
+            },
+        },
+    );
+}
+
+/**
+ * Returns a FunctionAssertion for `moveStake` which assumes the input amount exceeds the moveable
+ * amount of stake. Checks that the transaction reverts.
+ */
+export function moveStakeInvalidAmountAssertion(
+    deployment: DeploymentManager,
+): FunctionAssertion<[StakeInfo, StakeInfo, BigNumber], void, void> {
+    return new FunctionAssertion<[StakeInfo, StakeInfo, BigNumber], void, void>(
+        deployment.staking.stakingWrapper,
+        'moveStake',
+        {
+            after: async (_beforeInfo: void, result: FunctionResult) => {
+                // Ensure that the tx reverted.
+                expect(result.success).to.be.false();
+
+                // This isn't ideal but unfortunately there are several different revert errors that
+                // can be triggered by trying to move more stake than possible.
+                expect(
```

Comment:
Maybe just remove this assertion entirely since it seems like it could be a headache to maintain.

Reply:
🤷‍♂ works for me